### PR TITLE
Add extract sript

### PR DESCRIPTION
**Pull request by JoakimLandberg**
Adds a script to extract tar files. Also modified billing deploy
script to use the extract script.

### CODE REVIEW

File: deploy-billing [hunk 1]
```diff
@@ -1,68 +1,35 @@
...
-    setLink
-else
-    #New version
-    createDir
-    extract
-    setLink
-fi
+deleteDir
+createDir
+extract-mt "$file" "$target"
```
Comment (TheLudd) [inline deploy-billing]: This expects the function to be available globally right? 
Is it not better to just import the extract file?

File: extract-mt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+fi
+
+function extract {
+    cd "$workDir"
+    echo "Extracting $file into $target from `pwd`"
+    chown -R mediatool "$target"
+    sudo su -s /bin/bash -c "tar -C $target --strip-components=1 -xzf $file" mediatool
+}
+
+extract
```
Comment (TheLudd) [inline extract-mt]: This could be removed if the extract function were to be imported and called from other files

File: extract-mt [hunk 3]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+ROOT_UID=0
+workDir=`pwd`
+
+if [ $UID != $ROOT_UID ]; then
```
Comment (TheLudd) [inline extract-mt]: Not sure if this is needed in a util file.

**Comment by JoakimLandberg** [comment]
@TheLudd Imported extract-function in billing script and moved all code in the extract-mt script to the extract function to avoid name collisions.

File: deploy-billing [hunk 4]
```diff
@@ -1,68 +1,37 @@
 #!/bin/bash
-storageDir=/var/mediatool/
+target=/var/mediatool/mt-billing
 ROOT_UID=0
 
+source /var/mediatool-scripts/extract-mt
```
Comment (TheLudd) [inline deploy-billing]: Could we also make the imports relative? Like this http://stackoverflow.com/a/192305/1379657

**Comment by JoakimLandberg** [comment]
I looked at this earlier also. Doesn't seem to be any perfect way to do this, lots of caveats and gotchas. However, I changed script to include the extract function relative to the billing script. It will follow any symlinks etc we have for the script so it should be fine for us generally speaking.

**Pull request merged**